overlord/assertstate/assertmgr.go

```diff
@@ -153,3 +155,67 @@ func doValidateSnap(t *state.Task, _ *tomb.Tomb) error {
 	// TODO: set DeveloperID from assertions
 	return nil
 }
+
+func doValidateComponent(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	compsup, snapsup, err := snapstate.TaskComponentSetup(t)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot obtain snap setup: %s", err)
+	}
+
+	sha3_384, compSize, err := asserts.SnapFileSHA3_384(compsup.CompPath)
+	if err != nil {
+		return err
+	}
+
+	deviceCtx, err := snapstate.DeviceCtx(st, t, nil)
+	if err != nil {
+		return err
+	}
+
+	modelAs := deviceCtx.Model()
+
+	// TODO: i think that the provenance of the component should be the same as
+	// the snap's provenance
+	expectedProv := snapsup.ExpectedProvenance
+
+	err = doFetch(st, snapsup.UserID, deviceCtx, nil, func(f asserts.Fetcher) error {
+		if err := snapasserts.FetchComponentAssertions(f, snapsup.SideInfo, compsup.CompSideInfo, sha3_384, expectedProv); err != nil {
+			return err
+		}
+
+		// TODO: do we want this part here? it happens in doValidateSnap
+		// fetch store assertion if available
+		if modelAs.Store() != "" {
+			err := snapasserts.FetchStore(f, modelAs.Store())
+			if notFound, ok := err.(*asserts.NotFoundError); ok {
+				if notFound.Type != asserts.StoreType {
+					return err
+				}
+			} else if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+	if errors.Is(err, &asserts.NotFoundError{}) {
+		return fmt.Errorf("cannot find supported signatures to verify component %q and its hash (%v)", compsup.ComponentName(), err)
+	}
+	if err != nil {
+		return err
+	}
+
+	db := DB(st)
+	err = snapasserts.CrossCheckResource(compsup.ComponentName(), sha3_384, expectedProv, compSize, compsup.CompSideInfo, snapsup.SideInfo, modelAs, db)
+	if err != nil {
+		return err
+	}
+
+	// TODO: do we need to do something like snapasserts.CheckProvenanceWithVerifiedRevision for components?
```

Comment:
This seems like its already included in the checks above, but maybe I'm misunderstanding

Reply:
@pedronis can you speak to the necessity of this check and the one in https://github.com/snapcore/snapd/pull/13964/files#r1604780776? They're modeled after the ones from doValidateSnap

Reply:
this is about checking provenance value from inside the blob, we need to check with snapcraft whether it's being stored atm or not, it might not yet be but it should though

Reply:
Per @mr-cal, the snapcraft side of this isn't implemented at the moment

Reply:
Snapcraft takes the `provenance` keyword from a `snapcraft.yaml` and writes it into `meta/snap.yaml`.  

If I understand this correctly, snapcraft should always write the same provenance keyword in `meta/component.yaml` file?

Reply:
Yes, I believe that is correct.

Reply:
yes, correct

Reply:
From discussions with @pedronis, we're gonna leave the TODO for now and come back to this when the snapcraft side is implemented